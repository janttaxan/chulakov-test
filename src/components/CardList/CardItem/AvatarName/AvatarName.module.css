.avatarName {
  display: flex;
  align-items: center;
}

.image {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 8px;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  background-color: var(--secondary-100);
}

.image svg {
  height: 100%;
}

.name {
  font-size: 18px;
  font-weight: 500;
}

@media all and (max-width: 540px) {
  .name {
    font-size: 16px;
  }

  .image {
    margin-right: 6px;
    width: 40px;
    height: 40px;
  }
}
