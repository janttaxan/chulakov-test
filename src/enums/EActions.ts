export enum EActions {
  CARD_LIST_REQUEST = 'CARD_LIST_REQUEST',
  CARD_LIST_REQUEST_SUCCESS = 'CARD_LIST_REQUEST_SUCCESS',
  CARD_LIST_REQUEST_ERROR = 'CARD_LIST_REQUEST_ERROR',
  ADD_TO_FAVORITES = 'ADD_TO_FAVORITES',
  REMOVE_FROM_FAVORITES = 'REMOVE_FROM_FAVORITES',
}
