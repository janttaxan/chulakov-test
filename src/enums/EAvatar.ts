export enum EAvatar {
  sheep = 'sheep',
  raccoon = 'raccoon',
  pig = 'pig',
  penguin = 'penguin',
  owl = 'owl',
  lion = 'lion',
  koala = 'koala',
  fox = 'fox',
  dog = 'dog',
  cat = 'cat',
}
