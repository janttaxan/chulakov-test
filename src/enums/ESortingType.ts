export enum ESortingType {
  decrease = 'decrease',
  increase = 'increase',
}
