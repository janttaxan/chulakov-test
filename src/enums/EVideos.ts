export enum EVideos {
  boy = 'boy',
  shoe = 'shoe',
}
