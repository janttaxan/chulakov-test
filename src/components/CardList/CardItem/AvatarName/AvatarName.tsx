import styles from './AvatarName.module.css';
import { EAvatar } from '../../../../enums/EAvatar';
import {
  ImageSheep,
  ImageRaccoon,
  ImagePig,
  ImagePenguin,
  ImageOwl,
  ImageLion,
  ImageKoala,
  ImageFox,
  ImageDog,
  ImageCat,
} from '../../../Icons';

function getAvatar(avatar: EAvatar): JSX.Element {
  switch (avatar) {
    case EAvatar.sheep:
      return <ImageSheep/>;
    case EAvatar.raccoon:
      return <ImageRaccoon/>;
    case EAvatar.pig:
      return <ImagePig/>;
    case EAvatar.penguin:
      return <ImagePenguin/>;
    case EAvatar.owl:
      return <ImageOwl/>;
    case EAvatar.lion:
      return <ImageLion/>;
    case EAvatar.koala:
      return <ImageKoala/>;
    case EAvatar.fox:
      return <ImageFox/>;
    case EAvatar.dog:
      return <ImageDog/>;
    case EAvatar.cat:
      return <ImageCat/>;
  }
}

interface IAvatarNameProps {
  image: EAvatar;
  name: string;
}

export const AvatarName = ({ image, name }: IAvatarNameProps) => (
  <div className={styles.avatarName}>
    <div className={styles.image}>
      {getAvatar(image)}
    </div>
    <div className={styles.name}>{name}</div>
  </div>
);
