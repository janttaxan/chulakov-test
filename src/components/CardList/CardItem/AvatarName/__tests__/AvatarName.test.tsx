import React from 'react';
import { shallow } from 'enzyme';
import { AvatarName } from '../AvatarName';
import { EAvatar } from '../../../../../enums/EAvatar';

const setUp = (avatar: EAvatar, name: string = 'John Doe') => {
  switch (avatar) {
    case EAvatar.cat:
      return shallow(<AvatarName image={EAvatar.cat} name={name}/>);
    case EAvatar.dog:
      return shallow(<AvatarName image={EAvatar.dog} name={name}/>);
    case EAvatar.fox:
      return shallow(<AvatarName image={EAvatar.fox} name={name}/>);
    case EAvatar.koala:
      return shallow(<AvatarName image={EAvatar.koala} name={name}/>);
    case EAvatar.lion:
      return shallow(<AvatarName image={EAvatar.lion} name={name}/>);
    case EAvatar.owl:
      return shallow(<AvatarName image={EAvatar.owl} name={name}/>);
    case EAvatar.penguin:
      return shallow(<AvatarName image={EAvatar.penguin} name={name}/>);
    case EAvatar.pig:
      return shallow(<AvatarName image={EAvatar.pig} name={name}/>);
    case EAvatar.raccoon:
      return shallow(<AvatarName image={EAvatar.raccoon} name={name}/>);
    case EAvatar.sheep:
      return shallow(<AvatarName image={EAvatar.sheep} name={name}/>);
  }
};

describe('AvatarName', () => {
  test('should render with "Cat" avatar', () => {
    const component = setUp(EAvatar.cat);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Dog" avatar', () => {
    const component = setUp(EAvatar.dog);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Fox" avatar', () => {
    const component = setUp(EAvatar.fox);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Koala" avatar', () => {
    const component = setUp(EAvatar.koala);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Lion" avatar', () => {
    const component = setUp(EAvatar.lion);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Owl" avatar', () => {
    const component = setUp(EAvatar.owl);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Penguin" avatar', () => {
    const component = setUp(EAvatar.penguin);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Pig" avatar', () => {
    const component = setUp(EAvatar.pig);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Raccoon" avatar', () => {
    const component = setUp(EAvatar.raccoon);
    expect(component).toMatchSnapshot();
  });

  test('should render with "Sheep" avatar', () => {
    const component = setUp(EAvatar.sheep);
    expect(component).toMatchSnapshot();
  });
});
