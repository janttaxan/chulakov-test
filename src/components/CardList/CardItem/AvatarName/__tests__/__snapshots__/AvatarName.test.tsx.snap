// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarName should render with "Cat" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageCat />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Dog" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageDog />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Fox" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageFox />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Koala" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageKoala />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Lion" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageLion />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Owl" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageOwl />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Penguin" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImagePenguin />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Pig" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImagePig />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Raccoon" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageRaccoon />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;

exports[`AvatarName should render with "Sheep" avatar 1`] = `
<div
  className="avatarName"
>
  <div
    className="image"
  >
    <ImageSheep />
  </div>
  <div
    className="name"
  >
    John Doe
  </div>
</div>
`;
